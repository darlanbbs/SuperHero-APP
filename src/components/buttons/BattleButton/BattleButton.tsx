import { useForm } from "./../../../context/BattleContext";
import { Button } from "@mui/material";
import React from "react";

interface PropsBattle {
  id: number;
}

const BattleButton = ({ id }: PropsBattle) => {
  const { useContext, setUseContext } = useForm();

  const openToast = () => {
    setUseContext((prevState: any) => ({
      ...prevState,
      open: true,
    }));
    if (useContext.playerOne === 0) {
      setUseContext((prevState: any) => ({
        ...prevState,
        playerOne: id,
      }));
    }

    if (useContext.playerOne !== 0) {
      setUseContext((prevState: any) => ({
        ...prevState,
        playerTwo: id,
      }));
    }
    console.log(useContext);
  };

  return (
    <Button variant="contained" color="error" onClick={openToast}>
      Battle
    </Button>
  );
};

export default BattleButton;
