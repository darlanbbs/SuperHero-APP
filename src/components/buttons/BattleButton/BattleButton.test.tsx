import { render, screen } from "@testing-library/react";
import BattleButton from "./BattleButton";

describe("<BattleButton />", () => {
  const id = 1;
  it("should render the button", () => {
    // renderiza o component
    const { container } = render(<BattleButton id={id} />);

    // busca o elemento e verifica a existÃªncia dele
    expect(screen.getByRole("button", { name: /Battle/i })).toBeInTheDocument();
  });
});
