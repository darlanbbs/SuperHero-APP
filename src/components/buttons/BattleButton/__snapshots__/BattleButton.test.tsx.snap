// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BattleButton /> should render the button 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium css-1sszw6i-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Battle
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;

exports[`<BattleButton /> should render the heading 1`] = `
<button
  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedError MuiButton-sizeMedium MuiButton-containedSizeMedium css-1sszw6i-MuiButtonBase-root-MuiButton-root"
  tabindex="0"
  type="button"
>
  Battle
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
